lib/main.dart

```diff
@@ -1,20 +1,80 @@
 import 'package:flutter/material.dart';
+import 'package:shared_preferences/shared_preferences.dart';
 
-void main() {
-  runApp(const MainApp());
+SharedPreferences? prefs;
+
+void main() async {
+    WidgetsFlutterBinding.ensureInitialized();
+    prefs = await SharedPreferences.getInstance();
```

Comment:
I would prefer this to happen in the app but flutter seems to handle a loading screen quite nicely so this is okay for now